mstm/__init__.py

```diff
@@ -51,99 +51,109 @@ def separate_exponent(num):
+        InputFile.close()
+        
+        
+        # run MSTM
+        cmd = ['./mstm_ubuntu.exe', 'mstm.inp']
+        subprocess.check_call(cmd, cwd=temp_dir)
+        
+        # Go into results file
+        result_file = glob.glob(os.path.join(temp_dir,'mstm_out.dat'))[0]
+        print temp_dir
+        # Read correct lines of results file
+        mstm_result = iter(open(result_file))
+        S11 = []
+        S12 = []
+        S13 = []
+        S14 = []
+        S11col = -1
+        for line in mstm_result:
+            print line
             l = line.split()
-        if len(l)>0 and S11col>-1:
-            if 'matrix' in l:
-                break
-            if '************' in l[0]:
-                break
-            S11.append(float(l[S11col]))
-            S12.append(float(l[S11col + 1]))
-            S13.append(float(l[S11col + 2]))
-            S14.append(float(l[S11col + 3]))
-    Srow1 = [S11, S12, S13, S14]
-
-    os.chdir(current_directory)
-    if delete:
-        shutil.rmtree(temp_dir)
-
-    return Srow1, th, phi
+            if '11' in l:
+                S11col = l.index('11')
+                line = next(mstm_result)
+                l = line.split()
+            if len(l)>0 and S11col>-1:
+                if 'matrix' in l:
+                    break
+                if '************' in l[0]:
+                    break
+                S11.append(float(l[S11col]))
+                S12.append(float(l[S11col + 1]))
+                S13.append(float(l[S11col + 2]))
+                S14.append(float(l[S11col + 3]))
+        Srow1 = [S11, S12, S13, S14]
+        scat_mat_dict[wl] = [Srow1,th,phi]        
+        os.chdir(current_directory)
+        if delete:
+            shutil.rmtree(temp_dir)
+
+    return scat_mat_dict
 
 def calc_Is(targ, inc, th_min, th_max, phi_min, phi_max):
-    Srow1,th,phi = MSTM.calc_scat_matrix(self,targ, inc, th_min, th_max, phi_min, phi_max)
-    prefactor = 1/((2*np.pi*targ.index_matrix/inc.wavelength)**2)
-    #calculate Is using scattering matrix
-    # still need to fix this
-    Is = []#(Srow1[0]*inc.stokes_vec[0] + Srow1[1]*inc.stokes_vec[1] + Srow1[2]*inc.stokes_vec[2] + Srow1[3]*inc.stokes_vec[3])*prefactor*np.sin(th*np.pi/180)
-    return Is
-
-# create target class
+    scat_mat_dict = calc_scat_matrix(targ, inc, th_min, th_max, phi_min, phi_max)
+    Is_dict = dict.fromkeys(inc.wavelength)
+    for wl in inc.wavelength:
+        Is =[]
+        Srow1 = scat_mat_dict[wl][0]
+        th = scat_mat_dict[wl][1]
+        prefactor = 1/((2*np.pi*targ.index_matrix/wl)**2)
+        #calculate Is using scattering matrix
+        for i in range(0,len(th)):
+            Is.append((Srow1[0][i]*inc.stokes_vec[0] + Srow1[1][i]*inc.stokes_vec[1] + Srow1[2][i]*inc.stokes_vec[2] + Srow1[3][i]*inc.stokes_vec[3])*prefactor*np.sin(th[i]*np.pi/180))
+        Is_dict[wl] = Is
+    return Is_dict
+
+# class for target which will scatter incident light 
 class target:
```

Comment:
Try to follow [PEP 8](https://www.python.org/dev/peps/pep-0008/) guidelines for class names and variable names.